modules/bridge/src/main/ts/ephox/bridge/components/toolbar/ToolbarButton.ts

```diff
@@ -9,7 +9,7 @@ export interface BaseToolbarButtonSpec<I extends BaseToolbarButtonInstanceApi> {
   icon?: string;
   text?: string;
   onSetup?: (api: I) => (api: I) => void;
-  allowedModes?: string[];
+  allowedInReadonlyUiMode?: boolean;
```

Comment:
I'm a bit concerned over how this new option will impact future requirements to disable/enable buttons on other editor states like the selection. That is why I specced out the context thing that it would just be predicates that return true/false if the toolbar should be enabled or not. Those predicates could then run when the UI element is rendered and when the UI needs to be refreshed that could be on modeChange and nodeChange and be cached for performance. The readonly channel could then just be replaced with a update disabled state channel that just runs the predicate to determine the new state.
I guess one could implement that on top having this simply be mapped to `enabledContext: 'any'` or something since setting this new option to true means that it will always be enabled. But what would happen if you set both say `allowedInReadonlyUiMode: true` and `enabledContext: 'editable'` or something. Then you have two conflicting options.